controllers/vspheremachine_controller.go

```diff
@@ -90,12 +90,12 @@ func AddMachineControllerToManager(ctx context.Context, controllerManagerContext
 	}
 
 	if supervisorBased {
-		r.VMService = &vmoperator.VmopMachineService{Client: controllerManagerContext.Client}
 		networkProvider, err := inframanager.GetNetworkProvider(ctx, controllerManagerContext.Client, controllerManagerContext.NetworkProvider)
 		if err != nil {
 			return errors.Wrap(err, "failed to create a network provider")
 		}
 		r.networkProvider = networkProvider
```

Comment:
Optional: Do we need to duplicate r.networkProvider 🤔

Reply:
The r.networkProvider is utilized within the setVMModifiers function, where it invokes the ConfigureVirtualMachine interface of the network provider.